exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceRuntime.java

```diff
@@ -386,6 +389,66 @@ public void connectServiceApis(int[] serviceIds, Node node) {
     }
   }
 
+  /**
+   * Get service instance by its name.
+   *
+   * @throws IllegalArgumentException if there is no service with such name
+   */
+  public Service getServiceInstanceByName(String serviceName) {
+    synchronized (lock) {
+      return findService(serviceName)
+          .map(ServiceWrapper::getService)
+          .orElseThrow(() ->
+              new IllegalArgumentException("No service with such name in the Java runtime: "
+                  + serviceName));
+    }
+  }
+
+  /**
+   * Get service instance id by its name.
+   *
+   * @throws IllegalArgumentException if there is no service with such name
+   */
+  public int getServiceIdByName(String serviceName) {
+    synchronized (lock) {
+      return findService(serviceName)
+          .map(ServiceWrapper::getId)
+          .orElseThrow(() ->
+              new IllegalArgumentException("No service with such name in the Java runtime: "
+                  + serviceName));
+    }
+  }
+
+  /**
+   * Verifies that an Exonum raw transaction can be correctly converted to an executable
+   * transaction of given service.
+   *
+   * @param serviceId the id of the service
+   * @param txId the {@linkplain TransactionMessage#getTransactionId() transaction type identifier}
+   *     within the service
+   * @param arguments the {@linkplain TransactionMessage#getPayload() serialized transaction
+   *     arguments}
+   * @throws IllegalArgumentException if there is no service with such name in this runtime, or if
```

Comment:
actually there is a service **id** not **name** in the args.
How about null `arguments`, is it allowed?

Reply:
No, nulls are not, but I would say that's the conventional restriction :-)